flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/EmbeddedRocksDBStateBackend.java

```diff
@@ -521,6 +532,16 @@ private RocksDBOptionsFactory configureOptionsFactory(
                             ((ConfigurableRocksDBOptionsFactory) optionsFactory).configure(config);
                 }
                 LOG.info("Using configured options factory: {}.", optionsFactory);
+                if (DefaultConfigurableOptionsFactory.class.isAssignableFrom(clazz)) {
+                    LOG.warn(
+                            "{} is extending from {}, which is deprecated and will be removed in "
+                                    + "future. It is highly recommended to directly implement the "
+                                    + "ConfigurableRocksDBOptionsFactory without extending the {}. "
+                                    + "For more information, please refer to FLINK-24046.",
+                            optionsFactory,
+                            DefaultConfigurableOptionsFactory.class.getName(),
+                            DefaultConfigurableOptionsFactory.class.getName());
```

Comment:
My previous comment is that we might could use same method to check optionsFactory whether extending `DefaultConfigurableOptionsFactory` and print the warning both in line 502~507 and 535~543.

Reply:
Sure, I refined this method.